get_info.php

```diff
@@ -1 +1,200 @@
+    $stmt = $conn->prepare($query);
+
+    bind_variables($stmt, $_GET);
+    $stmt->execute();
+
+    if ($method == "GET") {
+        //fetch all results
+        $fetched = $stmt->fetchAll(PDO::FETCH_ASSOC);
+        $result['TYPE'] = $_GET["what"] . "_package";
+        $result['DATA'] = $fetched;
+    } else {
+        $result = set_http_response(202);
+    }
+
+    echo json_encode($result);
+} catch (PDOException $e) {
+    $result = set_http_response(500);
+    date_default_timezone_set('America/Toronto');
+    error_log(
+        date("Y-m-d h:i:sa") . " : " . $e->getMessage() . "\n",
+        3,
+        "errors.log"
+    );
+    print json_encode($result, JSON_PRETTY_PRINT);
+    exit();
+}
+
+/**
+ * This function handles GET requests and returns a sql query that gets the
+ * required data for the what according to the API documentation.
+ *
+ * @throws InvalidPage The requested what does not exist
+ * @return sql query to the corresponding what parameter
+ */
+function handle_get($parameters)
+{
+    switch ($parameters["what"]) {
+        case "tas":
+            return set_parameters(true, $parameters);
+            break;
+        case "questions":
+            return gen_query_questions();
+            break;
+        case "course_pairings":
+            return set_parameters(false, $parameters);
+            break;
+        case "user_info":
+            $include_photo = true;
+            if (isset($parameters["include_photo"])) {
+                if ($parameters["include_photo"] == 'false') {
+                    $include_photo = false;
+                }
+            }
+            $list_of_users = is_list_of_users($parameters["user_id"]);
+            return gen_query_user_info($include_photo, $list_of_users);
+            break;
+
+        default:
+            throw new Exception("InvalidPage");
+            break;
+    }
+}
+
+/**
+ * Returns true if user_id is a list of users.
+ * @param user_id Value of user_id parameters
+ * @return TRUE if user_id is a list
+ */
+function is_list_of_users($user_id)
+{
+    $user_id_list = explode(",", $user_id);
+
+    $list_of_users = false;
+    if (sizeof($user_id_list) > 1) {
+        $list_of_users = true;
+    }
+
+    return $list_of_users;
+}
+
+/**
+ * Binds variables to SQL query based on what parameter
+ *
+ * @param stmt Reference to PDO object that is a statement to be executed.
+ * @param parameters Array of parameters and values
+ */
+function bind_variables(&$stmt, $parameters)
+{
+    $stmt->bindParam(':user_id', $parameters["user_id"]);
+    if (
+        ($parameters["what"] == "course_pairings") ||
+        ($parameters["what"] == "tas")
+    ) {
+        if ($parameters["what"] == "course_pairings") {
+            if (isset($parameters["column"])) {
+                $stmt->bindParam(':column', $parameters["column"]);
+            }
+        } else {
+            $ta = 'ta';
+            $stmt->bindParam(':column', $ta);
```

Comment:
Why did you set a variable to use it only once?

Reply:
bindParam needs a reference. Passing in a string will cause an error.

Reply:
Okay. Put a comment to that effect so it's not confusing.

Reply:
The new schema requires me to change this so it is no longer a problem.